git_connector.py

```diff
@@ -485,10 +504,26 @@ def _clone_repo(self, param):
 
         action_result = self.add_action_result(ActionResult(dict(param)))
 
-        repo_dir = self.app_state_dir / self.repo_name
+        if not self.get_config().get("repo_uri") and not param.get("repo_url", None):
+            msg = "You must either provide a URL to clone or configure the app with information"
+            return action_result.set_status(phantom.APP_ERROR, msg)
+
+        if not param.get("repo_url", None):
+            repo_dir = self.app_state_dir / self.repo_name
+        else:
+            self.modified_repo_uri = param.get("repo_url")
+            self.branch_name = param.get("branch", "main")
+            folder_name = (
```

Comment:
Thanks for the changes.

It seems like we will not be able to clone different branch of same repo as the folder name remains same, giving error message "Repo already exists".